### DISCUSSION TITLE
Help solving this error

### DISCUSSION THREAD

**Discussion by SuryaSundarVadali** [Cyfrin/foundry-full-course-cu#2144] [Q&A] [answered]
![doubt_error](https://github.com/user-attachments/assets/74a707c8-efe3-4a42-b969-0c1e84c29cea)
error for override : Function has override specified but does not override anything.solidity(7792)
error for checkupkeep("") : Invalid type for argument in function call. Invalid implicit conversion from literal_string "" to bytes calldata requested.
tried changing bytes calldata to bytes memory still hasn't resolved

**Comment by EngrPips** [comment, marked as answer]
Hello @SuryaSundarVadali, Please remove those `override` keywords and let's see if the error persist and I have my `checkUpKeep` and `performUpKeep` function as the below 

```js
function checkUpkeep(bytes memory /* checkData */ )
        public
        view
        returns (bool upkeepNeeded, bytes memory /* performData */ )
    {
        // We don't use the checkData in this example. The checkData is defined when the Upkeep was registered.
        bool timeHasPassed = block.timestamp - s_lastTimeStamp >= i_interval;
        bool hasPlayer = s_players.length > 0;
        bool hasBalance = address(this).balance > 0;
        bool isOpen = s_raffleState == RaffleState.OPEN;

        upkeepNeeded = (timeHasPassed && hasPlayer && hasBalance && isOpen);
        return(upkeepNeeded, "0x0");
    }

    function performUpkeep(bytes calldata /* performData */) public {
        (bool upkeepNeeded,) = checkUpkeep("");
        if(!upkeepNeeded) revert Raffle__LotteryDoesNotNeedUpdate();
        s_raffleState = RaffleState.CALCULATING;
        console2.log("The address of the lottery contract is: ", address(this));
        uint256 requestId = i_vrfCoordinatorContract.requestRandomWords(VRFV2PlusClient.RandomWordsRequest({
            keyHash: i_gasLane,
            subId: i_subscriptionId,
            requestConfirmations: REQUEST_CONFIRMATIONS,
            callbackGasLimit: CALLBACK_GAS_LIMIT,
            numWords: NUM_OF_WORDS,
            // Set nativePayment to true to pay for VRF requests with Sepolia ETH instead of LINK
            extraArgs: VRFV2PlusClient._argsToBytes(VRFV2PlusClient.ExtraArgsV1({nativePayment: false}))
        }));
        emit Raffle__randomNumberRequested(requestId);
    }
```

**Reply by SuryaSundarVadali** [reply]
Yeah, it is fixed can you please explain why we should remove override
Thanks a lot for your help

**Reply by EngrPips** [reply]
Normally, we use the `override` keyword for a function that was flagged `virtual` in its original implementation, which we inherit from. That is not the case for both the `checkUpkeep` and `performUpkeep` functions as they are functions that we declared ourselves in our contract, but in the case of the `fulfillRandomWords` function, we inherited it from the `VRFConsumerBaseV2Plus` contract, which declared it and flagged it as `virtual` so we can `override` the implementation in our contract. The below image shows the `fulfillRandomWords` function inside of `VRFConsumerBaseV2Plus` 

<img width="1440" alt="Screenshot 2024-07-31 at 11 42 55" src="https://github.com/user-attachments/assets/7bfa3974-c8ed-4bd7-9419-36317d13a24a">

**Reply by SuryaSundarVadali** [reply]
Understood, thanks for your explanation!!

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing it.
